lib/tbot/config/migrate_test.go

```diff
@@ -0,0 +1,1132 @@
+/*
+Copyright 2023 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+import (
+	"bytes"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/lib/tbot/botfs"
+)
+
+func TestMigrate(t *testing.T) {
+	tests := []struct {
+		name       string
+		input      string
+		wantError  string
+		wantOutput *BotConfig
+	}{
+		{
+			name: "very full config",
+			input: `
+auth_server: example.teleport.sh:443
```

Comment:
nit: Worth popping this input into an `embed`? Same for the rest.

I don't feel strongly about this, so feel free to say no.

Reply:
Mmm - I know this file is long because of these, but it's much easier to read & write the tests when you can see the v1 config string and the wanted output value in the same spot. I feel like it'd kinda be a pain to go find the file.